docs/developer/payments.mdx

```diff
@@ -1131,6 +1131,172 @@ In case of a failure result, the [errors](api-reference/payments/objects/payment
 The mutation will trigger the [STORED_PAYMENT_METHOD_DELETE_REQUESTED](api-reference/webhooks/enums/webhook-event-type-sync-enum.mdx#code-style-fontweight-normal-webhookeventtypesyncenumbstored_payment_method_delete_requestedbcode) webhook.
 For more details about this webhook, please refer to its [documentation](developer/extending/webhooks/synchronous-events/stored-payment-method.mdx#delete-stored-payment-method-requested).
 
+### Tokenizing a new payment method
+
+:::info
+
+This feature was introduced in **Saleor 3.16**.
+
+:::
+
+:::caution
+
+This feature is in the **Feature Preview** stage, which means that it is available for experimentation and
+feedback. However, it is still undergoing development and is subject to modifications.
+
+:::
+
+:::info
+This feature is dedicated to [third-party](api-reference/apps/enums/app-type-enum.mdx#code-style-fontweight-normal-apptypeenumbthirdpartybcode) apps.
+:::
+
+:::info
+
+Tokenizing a payment method can also be done during the checkout process. Please refer to your payment app's documentation for details.
+
+:::
+
+Customers can request to store their payment method for later use. The tokenization process is handled by the payment app subscribed to synchronous webhooks. The process consists of three steps: initialization of the payment gateway (if required), tokenizing the payment method, and handling additional actions (if required). Each step has its dedicated mutation and webhook, which are described below.
+
+The diagram illustrates the general flow of tokenizing a payment method. The process may vary slightly for specific payment providers; for example, some payment providers may not require the initialization of the payment gateway.
+
+```mermaid
+sequenceDiagram
```

Comment:
<img width="828" alt="CleanShot 2023-09-07 at 13 05 55@2x" src="https://github.com/saleor/saleor-docs/assets/4144459/11ca793d-5bfc-4c4e-8805-448b9c19b4b5">

I don't get this part of the diagram - why should storefront use `paymentGatewayInitialize` before `paymentMethodInitializeTokenization`? I think that `paymentMethodInitializeTokenization` was supposed to be a counterpart of `paymentGatewayInitialize` for tokenization flow, so that you don't have to use the latter.

IMHO the app should return all necessary data for Payment Provider SDK (like drop-in) like `clientKey` or in case of Stripe initialize `SetupIntent`

In the later part of the docs, we have this paragraph:
> The [paymentGatewayInitializeTokenization](https://saleor-docs-git-programatic-payments-toke-b12f08-saleorcommerce.vercel.app/docs/next/api-reference/payments/mutations/payment-gateway-initialize-tokenization) mutation should be used when the payment gateway needs to be initialized first. For example, when retrieving the data required to initialize the drop-in on the storefront side.

Reply:
ok, nvm, I read the diagram one more time and it's fine, just please change `paymentGatewayInitialize` -> `paymentGatewayInitializeTokenization` ;)

Reply:
Fixed